ext/libxml/ruby_xml_node.c

```diff
@@ -45,13 +45,18 @@ static void rxml_node_deregisterNode(xmlNodePtr xnode)
   /* Has the node been wrapped and exposed to Ruby? */
   if (xnode->_private)
   {
+    /* Does it belong to another libxml user? */
+    if (!OWNED(xnode))
```

Comment:
Will take a look today and see what I come up with. @abrasive wrote the original patch (I'm just pushing it upstream and maintaining it internally) so he's best to answer re: macros vs. function calls.